### PR TITLE
refactor(core): get sign-in experience

### PR DESCRIPTION
**Pull request by IceHe**
## Summary

- feat(core): getEnabledSocialConnectorIds
- test(core): getEnabledSocialConnectorIds
- refactor(core): get sign-in experience
- test(core): get sign-in experience

## Linear Issue Reference

LOG-1939

## Testing

image
image

---
@logto-io/eng

### CODE REVIEW

File: packages/core/src/routes/sign-in-experience.ts [hunk 1]
```diff
@@ -16,7 +17,16 @@ export default function signInExperiencesRoutes<T extends AuthedRouter>(router:
    */
   router.get('/sign-in-exp', async (ctx, next) => {
     const signInExperience = await findDefaultSignInExperience();
-    ctx.body = signInExperience;
+    const { socialSignInConnectorIds: selectedSocialSignInConnectorIds } = signInExperience;
+    const enabledSocialConnectorIds = new Set(await getEnabledSocialConnectorIds());
```
Comment (simeng-li) [inline packages/core/src/routes/sign-in-experience.ts:21]: why do we net a set here?

**Review by simeng-li — CHANGES_REQUESTED** [review]

**Review by darcyYe — APPROVED** [review]

File: packages/core/src/routes/sign-in-experience.ts (hunk 1 above)
Comment (IceHe) [reply packages/core/src/routes/sign-in-experience.ts:20]: `set.has()` is better than `list.has` in performance.

```ts
    const socialSignInConnectorIds = selectedSocialSignInConnectorIds.filter((id) =>
      enabledSocialConnectorIds.has(id)
    );
```

Comment (simeng-li) [inline packages/core/src/routes/sign-in-experience.ts:21]: Promise.All?  findDefaultSignInExperience and getEnabledSocialConnectorIds can run simultaneously

Comment (IceHe) [reply packages/core/src/routes/sign-in-experience.ts]: Improved, PTAL.

**Review by simeng-li — APPROVED** [review]
LGTM

**Review by wangsijie — APPROVED** [review]

**Pull request merged**